assets/js/invidious_iframe_api.js

```diff
@@ -0,0 +1,386 @@
+                                break;
+                            case 'play':
+                                this.player_status = 1;
+                                break;
+                            case 'timeupdate':
+                                this.player_status = 1;
+                                this.eventdata = Object.assign({},this.eventdata,message.data.value);
+                                break;
+                            case 'pause':
+                                this.player_status = 2;
+                                break;
+                            case 'waiting':
+                                this.player_status = 3;
+                                break;
+                            case 'loadedmetadata':
+                                this.eventdata = Object.assign({},this.eventdata,message.data.value);
+                                break;
+                        }
+                        if(previous_status!==this.player_status){
+                            this.event_executor('statechange');
+                        }
+                    }
+            }
+        }
+    }
+    promise_resolve(){
+        var res_outer;
+        var pro = new Promise((res,rej)=>{res_outer=res});
+        return {'promise':pro,'resolve':res_outer};
+    }
+    promise_send_event(event_name){
+        if(invidious_embed.api_promise){
+            var pro_object = this.promise_resolve();
+            this.message_wait[event_name].push(pro_object.resolve);
+            this.postMessage({eventname:event_name});
+            return pro_object.promise;
+        }
+        else{
+            return this.eventdata[event_name];
+        }
+    }
+    getPlayerState(){
+        return this.player_status;
+    }
+    playVideo(){
+        this.postMessage({eventname:'play'});
+    }
+    pauseVideo(){
+        this.postMessage({eventname:'pause'});
+    }
+    getVolume(){
+        return this.promise_send_event('getvolume');
+    }
+    setVolume(volume){
+        volume = Number(volume);
+        this.volume = volume;
+        if(volume!==NaN&&volume!=undefined&&volume>=0&&volume<=100){
+            this.postMessage({eventname:'setvolume',value:volume/100});
+        }
+    }
+    getIframe(){
+        return this.player_iframe;
+    }
+    destroy(){
+        this.player_iframe.remove();
+    }
+    mute(){
+        this.postMessage({eventname:'setmutestatus',value:true});
+    }
+    unMute(){
+        this.postMessage({eventname:'setmutestatus',value:false});
+    }
+    isMuted(){
+        return this.promise_send_event('getmutestatus');
+    }
+    seekTo(seconds,allowSeekAhead){//seconds must be a number and allowSeekAhead is ignore
+        seconds = Number(seconds);
+        if(seconds!==NaN&&seconds!==undefined){
+            this.postMessage({eventname:'seek',value:seconds});
+        }
+    }
+    setSize(width,height){//width and height must be Number
+        this.player_iframe.width = Number(width);
```

Comment:
In some methods `Number` is controlled strictly, in others (like this) not. Please, unify parameters handling. Better using some private method.